wave_utils/src/config.cpp

```diff
@@ -0,0 +1,362 @@
+#include "wave/utils/config.hpp"
+
+
+namespace wave {
+
+ConfigParser::ConfigParser(void) {
+    this->configured = false;
+    this->loaded = false;
+}
+
+int ConfigParser::getYamlNode(std::string key, YAML::Node &node) {
+    int end;
+    std::string s;
+    std::vector<std::string> elements;
+    std::vector<YAML::Node> traversal;
+
+    // split key into elements
+    s = "";
+    for (int i = 0; i < key.length(); i++) {
```

Comment:
Does this function `getYamlNode()` not set `node = node["some_level_deeper"]` ?
Chances are my example is buggy : ) the point is you can do it in one legible loop.